app/controllers/rooms_controller.rb

```diff
@@ -13,15 +13,12 @@ def api_help
   	render nothing: true
   end
 
-
-  # Hey Britt, I haven't thought through this enough but will this
-  # hold for concurrency issues?  It seems like we only store one
-  # value in Redis here and it is possible if two people try to make
-  # the request at the same time, this could return one user's wolfram
-  # html to another user.  I could be mistaken here, could you let me know?
-  def get_wolfram
+  # this function returns a json object containing the wolfram/wiki html to put in the div, and the augmented api type
+  # calculated in rooms_helper wolfram and wiki query methods
+  def get_api_html
     hash = {}
-    hash['result'] = $redis.get("wolfram_html").to_s
+    hash['result'] = $redis.get("#{current_user.id}:api_html").to_s
```

Comment:
Now we won't have concurrency issues with this fix - unique values for every user.